netbox/data_source_netbox_vrfs.go

```diff
@@ -0,0 +1,149 @@
+package netbox
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/fbreckle/go-netbox/netbox/client"
+	"github.com/fbreckle/go-netbox/netbox/client/ipam"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/id"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func dataSourceNetboxVrfs() *schema.Resource {
+	return &schema.Resource{
+		Read:        dataSourceNetboxVrfsRead,
+		Description: `:meta:subcategory:IP Address Management (IPAM):`,
+		Schema: map[string]*schema.Schema{
+			"filter": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"value": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"limit": {
+				Type:             schema.TypeInt,
+				Optional:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.IntAtLeast(1)),
```

Comment:
Does it make sense to use the "at least 1" validator if the default is 0, i.e. not fulfilling the validator?

Reply:
Not sure I understand. Kept it consistent with limit in other bulk data sources (vlans / prefixes).

Reply:
True. I just tested it locally and for some reason the setup works as expected. The `IntAtLeast` validator only seems to run if the `limit` field is actually provided by the user. :shrug: